File: Modules/SEDO/EMS/SIE.Web.EMS.EquipMaint/Maintains/Plans/Commands/DeleteMaintainPlanExecuteAttachmentCommand.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
using SIE.Common.Attachments;
using SIE.Domain;
using SIE.Domain.Validation;
using SIE.EMS.Maintains.Controller;
using SIE.EMS.Maintains.Plans;
using SIE.EMS.Maintains.Records;
using SIE.MetaModel;
using SIE.Security;
using SIE.Web.Command;
using SIE.Web.Common.Attachments.Commands;
using SIE.Web.Core.Common.Commands;
using System;
using System.Collections.Generic;
using System.Text;

namespace SIE.Web.EMS.EquipMaint.Maintains.Plans.Commands
{
    /// <summary>
    /// 删除图片
    /// </summary>
    public class DeleteMaintainPlanExecuteAttachmentCommand : ImmediateDeleteCommand
    {

    }
}

File: Modules/SEDO/EMS/SIE.Web.EMS.EquipMaint/Maintains/Plans/Commands/DeleteMaintainPlanExecuteAttachmentCommand.js
SIE.defineCommand('SIE.Web.EMS.EquipMaint.Maintains.Plans.Commands.DeleteMaintainPlanExecuteAttachmentCommand', {
    extend: 'SIE.Web.Core.Common.Commands.ImmediateDeleteCommand',
    meta: { text: "删除", group: "edit", iconCls: "icon-Delete icon-blue" },
    canExecute: function (listview) {
        if (listview.getParent().getCurrent()) {
            var parentEntity = listview.getParent().getCurrent();
            var item = parentEntity;
            if (item.data.ExeState == 1 || item.data.ExeState == 3) {
                return false;
            }
            return true;
        }
        else {
            return false;
        }
    },
});